### PR TITLE
Refactor components and use effector to manage state

### PR DESCRIPTION
**Pull request by renjithgr**
This commit does two things mainly
1. Simplifies components. For example, instead of `Source` and `Target` now there is `CodeEditor`
2. Uses [effector](https://github.com/zerobias/effector) for managing the state.

### CODE REVIEW

**Comment by renjithgr** [comment]
As per our discussion, I removed the use of `effector` but kept the other changes I made.

File: src/components/code-editor/CodeEditor.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react';
+
+import './code-editor.scss';
+
+const CodeEditor = (props) => {
+    const { placeholder, initialValue, valueChanged, readOnly, type } = props;
+    const [value, setValue] = useState(initialValue);
+
+    useEffect(() => { setValue(initialValue); }, [initialValue]);
```
Comment (jamesgeorge007) [inline src/components/code-editor/CodeEditor.js]:
```suggestion
    useEffect(() => setValue(initialValue), [initialValue]);
```

File: src/components/select-box/SelectBox.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { useContext } from 'react';
+
+import './select.scss';
+import {Context} from "../../App";
```
Comment (jamesgeorge007) [inline src/components/select-box/SelectBox.js]:
```suggestion
import { Context } from "../../App";
```

File: src/components/title/Title.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React, {useContext} from 'react';
+
+import './title.scss';
+import {Context} from "../../App";
```
Comment (jamesgeorge007) [inline src/components/title/Title.js]:
```suggestion
import { Context } from "../../App";
```

File: src/components/header/Header.js [hunk 4]
```diff
@@ -1,11 +1,12 @@
-import React, { useContext } from 'react';
+import React, {useContext} from 'react';
```
Comment (jamesgeorge007) [inline src/components/header/Header.js]:
```suggestion
import React, { useContext } from 'react';
```

File: src/components/title/Title.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React, {useContext} from 'react';
```
Comment (jamesgeorge007) [inline src/components/title/Title.js]:
```suggestion
import React, { useContext } from 'react';
```

File: src/helpers/transform.js [hunk 6]
```diff
@@ -2,14 +2,25 @@ import postcss from 'postcss';
 import postcssJs from 'postcss-js';
 import transform from 'css-to-react-native';
 
+export const toJSSObject = (cssText) => {
```
Comment (jamesgeorge007) [inline src/helpers/transform.js]:
```suggestion
const toJSSObject = (cssText) => {
```

File: src/App.js [hunk 7]
```diff
@@ -1,30 +1,49 @@
...
-      value={{ state, setState, rawSnippet, currentSnippet, setSnippet, type, setType }}
-    >
-      <div className="wrapper">
-        <Header />
-        <Title />
-        <Transform />
-      </div>
-    </AppContext.Provider>
-  );
+    const [outputType, setOutputType] = useState('JSS');
```
Comment (jamesgeorge007) [inline src/App.js]: Let's go with `React Native`.

Comment (renjithgr) [reply src/App.js]: I made `React Native` the default option.

**Comment by renjithgr** [comment]
> We have an initial CSS snippet that is intended to show up initially so that the users can get an idea regarding the usage. Also, it should be the case if the `textarea` field was cleared (left empty).

This should be fixed now.

**Comment by jamesgeorge007** [comment]
Thanks

**Pull request merged**